it/uk/gov/hmrc/partnershipidentificationfrontend/connectors/RegistrationConnectorISpec.scala

```diff
@@ -42,11 +42,24 @@ class RegistrationConnectorISpec extends ComponentSpecHelper with RegisterStub {
 
     "return RegistrationFailed" when {
       "the registration has not been successful" in {
-        stubRegisterGeneralPartnership(testSautr, testRegime)(OK, RegistrationFailed)
+        stubRegisterGeneralPartnership(testSautr, testRegime)(OK, RegistrationFailed(Some(testRegistrationFailure)))
```

Comment:
There is a question here about passing an instance of RegistrationFailed to the stub. As the stub converts the instance into Json we are basically using our implicit "writes" method to do the conversion. It may be better to create a JsObject that represents the failed registration and pass this to the stub. I think we might need a team discussion on this.